protocol/contracts/beanstalk/sun/SeasonFacet/Distribution.sol

```diff
@@ -0,0 +1,154 @@
+                }
+            }
+            // If no cap exceeded, amounts are final.
+            if (!capExceeded) break;
+        }
+
+        // Ship it.
+        for (uint256 i; i < shipmentAmounts.length; i++) {
+            // Success is not verified here. Receive functions should be designed to never revert.
+            // address(this).call(
+            //     abi.encodeWithSelector(
+            //         bytes4(keccak256("receive(uint8,uint256,bytes)")),
+            //         shipmentRoutes[i].recipient,
+            //         shipmentAmounts[i],
+            //         shipmentRoutes[i].data
+            //     )
+            // );
+            receiveShipment(
+                shipmentRoutes[i].recipient,
+                shipmentAmounts[i],
+                shipmentRoutes[i].data
+            );
+        }
+
+        emit Ship(s.season.current, shipmentRoutes, shipmentAmounts);
+    }
+
+    /**
+     * @notice Gets the shipping plan for all shipping routes.
+     * @dev Determines which routes are active and how many Beans they will receive.
+     * @dev GetPlan functions should never fail/revert. Else they will have no Beans allocated.
+     */
+    function getShipmentPlans(
+        Storage.ShipmentRoute[] memory shipmentRoutes
+    ) private view returns (ShipmentPlan[] memory shipmentPlans, uint256 totalPoints) {
+        shipmentPlans = new ShipmentPlan[](shipmentRoutes.length);
+        for (uint i; i < shipmentRoutes.length; i++) {
+            (bool success, bytes memory returnData) = shipmentRoutes[i].planContract.staticcall(
+                abi.encodeWithSelector(shipmentRoutes[i].planSelector, shipmentRoutes[i].data)
+            );
+            if (success) {
+                shipmentPlans[i] = abi.decode(returnData, (ShipmentPlan));
+            } else {
+                shipmentPlans[i] = ShipmentPlan({points: 0, cap: 0});
+            }
+            totalPoints += shipmentPlans[i].points;
+        }
+    }
+    /**
+     * @notice Determines the amount of Beans to distribute to each shipping route based on points.
+     * @dev Does not factor in route cap.
+     * @dev If points are 0, does not alter the associated shippingAmount.
+     * @dev Assumes shipmentAmounts and shipmentRoutes have matching shape and ordering.
+     */
+    function getBeansFromPoints(
+        uint256[] memory shipmentAmounts,
+        ShipmentPlan[] memory shipmentPlans,
+        uint256 totalPoints,
+        uint256 beansToShip
+    ) private pure {
+        for (uint256 i; i < shipmentPlans.length; i++) {
+            // Do not modify amount for streams with 0 points. They either are zero or have already been set.
+            if (shipmentPlans[i].points == 0) continue;
+            shipmentAmounts[i] = (beansToShip * shipmentPlans[i].points) / totalPoints; // round down
+        }
+    }
+
+    //////////////////////////////////// SHIPPING PLAN GETTERS ////////////////////////////////////
```

Comment:
Could we put the shipping plan getters in a seperate facet? The main benefit is that the shipping plans can be updated without updating the season facets.

Reply:
They aren't worthy of a facet. Instead we could

- put them in a separate contract that is inherited into the season facet for cleaner code separation
or
- put them in an external contract that is outside beanstalk and immutable. then redeploy externally and update shipping route plan getter contract address on any future updates.

Reply:
i decided to place them in a contract that lives outside of beanstalk

https://github.com/BeanstalkFarms/Beanstalk/pull/860/commits/bbc0486ed0591c1b30cb75365c393b9efa03d353#diff-cf3f4898f41eee71e7b37cb1f5e642c532b4efd2ce1ee23de550f167e61ea6b5R25